components/brave_shields/browser/ad_block_subscription_service_manager.h

```diff
@@ -0,0 +1,133 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_SHIELDS_BROWSER_AD_BLOCK_SUBSCRIPTION_SERVICE_MANAGER_H_
+#define BRAVE_COMPONENTS_BRAVE_SHIELDS_BROWSER_AD_BLOCK_SUBSCRIPTION_SERVICE_MANAGER_H_
+
+#include <map>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "base/macros.h"
+#include "base/memory/scoped_refptr.h"
+#include "base/optional.h"
+#include "base/synchronization/lock.h"
+#include "base/values.h"
+#include "brave/components/adblock_rust_ffi/src/wrapper.h"
+#include "brave/components/brave_component_updater/browser/brave_component.h"
+#include "brave/components/brave_shields/browser/ad_block_subscription_service.h"
+#include "brave/components/brave_shields/browser/custom_subscription_download_manager.h"
+#include "chrome/browser/profiles/profile.h"
```

Comment:
please refresh https://www.chromium.org/developers/design-documents/cookbook
this dependency (and similar) are illegal, that's why `gn` complains